File: lib_common/src/main/java/com/style/data/db/FileDownloadStateDao.java
package com.style.data.db;

import androidx.room.*;
import com.style.data.fileDown.FileDownloadStateBean;

import java.util.ArrayList;
import java.util.List;

@Dao
public interface FileDownloadStateDao {
    @Query("SELECT * FROM file_download")
    public List<FileDownloadStateBean> getAll();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public long save(FileDownloadStateBean bean);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    public int update(FileDownloadStateBean entity);

    @Query("UPDATE file_download SET status=:status,downloadSize=:downloadSize WHERE url=:url")
    public int update(int status, int downloadSize, String url);
}

File: module_common_data/src/main/java/com/style/data/db/FileDownloadStateDao.kt
package com.style.data.db

import android.arch.persistence.room.*
import com.style.data.fileDown.FileDownloadStateBean

@Dao
interface FileDownloadStateDao {
    @Query("SELECT * FROM file_download")
    fun getAll(): List<FileDownloadStateBean>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun save(bean: FileDownloadStateBean): Long

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun update(entity: FileDownloadStateBean): Int

    @Query("UPDATE file_download SET status=:status,downloadSize=:downloadSize WHERE url=:url")
    fun update(status: Int, downloadSize: Int, url: String): Int
}